### PR TITLE
updated Docker files

### PR DESCRIPTION
**Pull request by jssmith**
This change updates Docker files. Changes include:
- Remove ray user and run as root instead (best practice for Docker)
- Use Anaconda Python distribution
- Update to new Ray version

### CODE REVIEW

File: docker/examples/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,6 @@
...
-# Tensorflow
-RUN pip install --upgrade https://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-0.9.0-cp27-none-linux_x86_64.whl
-
-# SciPy
-RUN pip install scipy
-
-# Gym
-RUN sudo apt-get -y install zlib1g-dev libjpeg-dev xvfb libav-tools xorg-dev python-opengl libsdl2-dev swig wget
+FROM ray-project/deploy
+RUN conda install -y -c conda-forge tensorflow
```
Comment (robertnishihara) [inline docker/examples/Dockerfile]: Just curious, why install tensorflow this way instead of via pip?

File: docker/base-deps/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# The deploy Docker image build a self-contained Ray instance suitable
+# for end users.
+
+FROM ubuntu:xenial
+RUN apt-get update
+RUN apt-get install -y vim git wget
+RUN apt-get install -y cmake build-essential autoconf curl libtool libboost-all-dev unzip
```
Comment (robertnishihara) [inline docker/base-deps/Dockerfile]: I believe best practice for Dockerfiles is to combine stuff into one big command and remove the temporary files, e.g., something like the following (taken from https://github.com/openai/gym/blob/master/Dockerfile)

```
RUN apt-get update \
    && apt-get install -y libav-tools \
    python-numpy \
    python-scipy \
    python-pyglet \
    python-setuptools \
    libpq-dev \
    libjpeg-dev \
    curl \
    cmake \
    swig \
    python-opengl \
    libboost-all-dev \
    libsdl2-dev \
    wget \
    unzip \
    git \
    xpra \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/* \
    && easy_install pip
```

File: docker/examples/Dockerfile (hunk 1 above)
Comment (jssmith) [inline docker/examples/Dockerfile]: This is the suggested command for installation on Anaconda given on the Tensorflow page:
https://www.tensorflow.org/get_started/os_setup#anaconda_installation

I had tried using the pip install but encountered errors.

File: docker/base-deps/Dockerfile (hunk 2 above)
Comment (jssmith) [inline docker/base-deps/Dockerfile]: Ok, I'm updating this. It looks like the temporary files removed are pretty small in this case (already dealing with Anaconda on one line) but I'd like to stick with the pattern.

**Pull request merged**